contracts/eosio.system/src/rex.cpp

```diff
@@ -353,6 +353,25 @@ namespace eosiosystem {
       }
    }
 
+   /**
+    * @brief Updates REX pool and deposits token to eosio.rex
+    *
+    * @param payer - the payer of donated funds.
+    * @param quantity - the quantity of tokens to donated to REX with.
+    * @param memo - the memo string to accompany the transfer.
+    */
+   void system_contract::donatetorex( const name& payer, const asset& quantity, const std::string& memo )
+   {
+      require_auth( payer );
+      check( rex_available(), "rex system is not initialized" );
+      check( quantity.symbol == core_symbol(), "quantity must be core token" );
+
```

Comment:
Perhaps also include `quanity.amount > 0` to be positive

This check will happen at the EOS token transfer, but best to check this prior to adding to REX pool.

"quantity must be positive"